docs/hosting-a-server-with-northstar/dedicated-server/hosting-on-linux.md

```diff
@@ -10,9 +10,194 @@ If you do not have a DirectX 11 capable GPU in your system (for example if you a
+#### Ubuntu
+1. Update `apt` package index and install packages to allow `apt` to use a repo over HTTPS
+```
+sudo apt-get update
+
+ sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+```
+
+2. Add Docker's official GPG key
+```
+sudo mkdir -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+```
+
+3. Use the following commmand to set up the repository
+```
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+```
+
+4. Update `apt` index with `sudo apt-get update` and then install docker engine, containerd, and docker compose.
+```
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+```
+
+5. Verify docker engine is installed by running this
+```
+sudo docker run hello-world
+```
+
+#### Debian
+1. Update `apt` package index and install packages to allow `apt` to use a repo over HTTPS
+```
+sudo apt-get update
+
+ sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+```
+
+2. Add Docker's official GPG key
+```
+sudo mkdir -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+```
+
+3. Use the following commmand to set up the repository
+```
+echo \
+  " echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+```
+
+4. Update `apt` index with `sudo apt-get update` and then install docker engine, containerd, and docker compose.
+```
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+```
+
+5. Verify docker engine is installed by running this
+```
+sudo docker run hello-world
+```
+
+#### Fedora
+Install the `dnf-plugins-core` package (which provides the commands to manage your DNF repositories) and set up the repository.
+```
+sudo dnf -y install dnf-plugins-core
+
+ sudo dnf config-manager \
+    --add-repo \
+    https://download.docker.com/linux/fedora/docker-ce.repo
+```
+
+Install the latest version of docker engine, containerd, and docker compose
+```
+sudo dnf install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+```
+If prompted to accept the GPG key, verify that the fingerprint matches `060A 61C5 1B55 8A7F 742B 77AA C52F EB6B 621E 9F35` (cross reference this with the [GPG key fingerprint provided by dockers docs](https://docs.docker.com/engine/install/fedora/)) , and if so, accept it.
+
+Start docker
+```
+sudo systemctl start docker
+```
+
+Then verify that docker is installed correctly by running this
+```
+sudo docker run hello-world
+```
```

Comment:
I'd rather not add instructions how to install Docker to this wiki, primarily cause they can change over time and that will require us keeping track of them and updating them when necessary.

Also as a sidenote, why do people recommend using custom repo for Docker (on Ubuntu). It's in the default repos via `apt install docker.io`

Reply:
I'd still remove the install guide. Just link to the official docs, otherwise it takes up a large proportion just for setting up Docker IMO

Reply:
eh, fair. I'll remove the docker install but still leave the install link in that case